src/rust/ide/src/controller/graph.rs

```diff
@@ -562,7 +581,13 @@ impl Handle {
         let destination_info         = self.destination_info(connection)?;
         let source_identifier        = source_info.target_ast()?.clone();
         let updated_target_node_expr = destination_info.set(source_identifier)?;
-        self.set_expression_ast(connection.destination.node, updated_target_node_expr)
+        self.set_expression_ast(connection.destination.node,updated_target_node_expr)?;
+
+        // Reorder node lines, so the connection target is after connection source.
+        // Actually this is needed only in some order-dependant context. Once we have better
+        // information about the graph's underlying monadic context, we might want to constrain
+        // this operation.
+        self.place_node_line_after(connection.source.node,connection.destination.node)
```

Comment:
So we sent notification about text change twice as I see. Can we do something with it?

If it is troublesome, add Fixme here for that.